### DISCUSSION TITLE
ShyFox 4.0-css teaser

### DISCUSSION THREAD

**Discussion by Naezr** [Naezr/ShyFox#131] [Announcements]
[Запись экрана от 2024-08-11 19-55-01.webm](https://github.com/user-attachments/assets/771cd1b0-4b79-4dcb-afa2-a2540b609ae7)

upd:

[Запись экрана от 2024-09-19 11-49-54.webm](https://github.com/user-attachments/assets/087e93cd-c5a7-49bc-82c8-c0d9cb94f877)

**Comment by semi710** [comment]
firefox should hire you buddy... this is sick ....

**Comment by loaykhalifa** [comment]
![CleanShot 20240812 - 01-41](https://github.com/user-attachments/assets/c7ded3c9-b515-4f83-8c29-9eb537eca16b)

Thanks for awesome theme, this suggest for URL

**Comment by Naezr** [comment]
New Firefox sidebar is cool

[Запись экрана от 2024-08-14 18-46-46.webm](https://github.com/user-attachments/assets/e323b75c-ae41-44a8-902c-98c30d73812e)

**Comment by Naezr** [comment]
Native vertical tabs now lack only one function: spaces. It’s still in development, so let’s hope for the best. Otherwise, the new sidebar available in Firefox Nightly has all the other important features that Sidebery gives. Except for spaces of course

**Comment by semi710** [comment]
Firefox new update integrating default vertical bar

**Comment by Naezr** [comment]
![image](https://github.com/user-attachments/assets/b6e25d52-1749-47e1-8e9b-eac3f3edd577)
![image](https://github.com/user-attachments/assets/762d6035-51b2-4040-bd7b-3b34e5989a97)

**Reply by Naezr** [reply]
I think it would be cool to have the ability to add some blurred image on the sidebar background

**Reply by semi710** [reply]
> I think it would be cool to have the ability to add some blurred image on the sidebar background

Make the wallpaper blurred there

**Reply by loaykhalifa** [reply]
Yes, yes, yes 
Second photo, the URL is awesome 
that make everything in sidebar (URL, TOOLBAR & TABS)
share beta version, I can test it in mac if you want.
I already run your theme in Firefox nightly.

![CleanShot 20240818 - 19-58](https://github.com/user-attachments/assets/e3bfc55e-fdd1-4cd4-93b2-a40eebb1176e)

**Reply by Naezr** [reply]
> share beta version

https://github.com/Naezr/ShyFox/tree/4.0-alpha

**Comment by Naezr** [comment]
[Запись экрана от 2024-08-18 12-44-14.webm](https://github.com/user-attachments/assets/eee547fe-c484-4bdb-b86d-062afbd30c44)

**Reply by loaykhalifa** [reply]
Can't show close & minimize button in alpha 4.0
even I do 

** - On macOS, you need to disable native context menus to see icons.
widget.macos.native-context-menus => false

![CleanShot 20240825 - 02-16 2](https://github.com/user-attachments/assets/ad36de57-ac59-4454-a61b-774385635c8f)
![CleanShot 20240825 - 02-17](https://github.com/user-attachments/assets/c57d31ff-171e-4cae-a6c4-81d9d7beb7eb)

**Reply by loaykhalifa** [reply]
can you add sidbary css in alpha too

**Reply by Naezr** [reply]
The window control buttons have not yet been implemented. First I need to make the left side autohide. The window control buttons have to adapt to everything around it. And when everything around is not done yet, there is no point in making them. 4.0 is being written from scratch, and this is just an early prototype, nothing more. Styles for Sidebery are easily ported from 3.0, as well as some other things. However, that's not as important now as the implementation of the core functionality. 

----------------------------------------------

> ** - On macOS, you need to disable native context menus to see icons. widget.macos.native-context-menus => false

It has nothing to do with alpha 4.0 at the moment. It doesn't have context menu styles and icons in it. Obviously that will come later, but there are more important things right now

**Reply by loaykhalifa** [reply]
![CleanShot 20240826 - 03-50](https://github.com/user-attachments/assets/3b2988ee-a7af-408c-8f09-f5703a14fdbd)

You main this issue ?!!

**Reply by Naezr** [reply]
Realize it's alpha. Nothing's finalized yet. Even the most basic things aren't done yet

**Comment by Naezr** [comment]
[Запись экрана от 2024-08-19 15-00-51.webm](https://github.com/user-attachments/assets/37a30d70-9a60-43e0-88be-2340fdb31078)

**Comment by armindarvish** [comment]
@Naezr Since you are working on 4.0, I think in addition to the native vertical tabs and sideberry, it would be nice if you could add support for other some other vertical tab bar extensions as well. I, personally, prefer the simplicity of "Tab Center Reborn" over the bells and whistles in sideberry and it takes less room on the right side and looks much cleaner too. 

I have modified shyfox in my own setup to do this, but with the complexity of the code you have, it's now hard to update every time. Here is a screenshot:

![Screen Recording 2024-09-05 at 3 47 37 AM](https://github.com/user-attachments/assets/70c18228-1de2-4c30-8e03-c83770f54992)

**Reply by Naezr** [reply]
Quite realizable

**Comment by loaykhalifa** [comment]
There are three spots 
![CleanShot 20240907 - 15-10](https://github.com/user-attachments/assets/e2028e0e-f8bc-42b7-98e1-5ac0caf2b575)
First spot: sidebar ON with new firefox functions (AI chatbot,....)

Second spot: can you hide this section? At all.
To make sideBERY wider like you see in second photo
![CleanShot 20240907 - 15-22](https://github.com/user-attachments/assets/f90cae66-3b4f-4f12-abca-a2a3d8b9b8f5)


Third spot: there is a cut-off of sideBERY on the bottom when turn ON (true)
- shyfox.position.navbar
- shyfox.position.bookmarkbar
![CleanShot 20240907 - 15-28](https://github.com/user-attachments/assets/7d4afc60-1918-4b77-a18c-00c1873b2340)

**Comment by loaykhalifa** [comment]
This is what I mean
![shyFox](https://github.com/user-attachments/assets/fa8390a1-a041-469a-bbd5-ab3f3973087b)

**Reply by Naezr** [reply]
That's roughly how I imagine the future “Sidebery mode” to look like. 

Since Firefox will have its own vertical tabs, some users will want to use them instead of Sidebery. It will also provide some benefits, like transparency and the ability to use the other sidebar applets.

For fans of Sidebery and other extensions with vertical tabs, there will be a special setting in 4.0. When this setting is enabled, if Sidebery (or another extension) is opened in the sidebar, then the rest of the sidebar interface will disappear. Like the current behavior in 3.0. If this setting is disabled, extensions with vertical tabs will behave like other sidebar extensions.

**Reply by 0xda157** [reply]
As of Firefox v130 Sidebery can be transparent. You just need to enable `browser.tabs.allow_transparent_browser` and add `window#webextpanels-window { background: transparent !important; }` to userChrome.

**Comment by loaykhalifa** [comment]
Nice work
sideBERY Heights fixed 
![CleanShot 20240907 - 18-56](https://github.com/user-attachments/assets/b59139a8-d36e-4cf2-8dbc-462e62d6574f)

**Reply by loaykhalifa** [reply]
The main webpage doesn't fold 
& sideBERY Heights fixed 👍 (bottom issue) 
![CleanShot 20240907 - 19-17](https://github.com/user-attachments/assets/6059acfd-b4b4-413e-863b-68d3b0befbec)

**Comment by loaykhalifa** [comment]
After last alpha update everything work smoothly thank you very much Naezr 
But I still prefer sideBERY functionality on photo no. [2]&[3]
Because SideBERY customizable  

> 
> **photo [1]**
> 
>   - sideBERY OFF
> Native nightly Firefox
>   - Sidebar ON
>   - Vertical tab ON 
> ![CleanShot 20240909 - 15-40](https://github.com/user-attachments/assets/8e097663-c0c7-4b7e-b145-e3ae336feb18)

_________________________________________________________

> **photo [2]**
> 
>   - sideBERY ON
> Native nightly Firefox
>   - Sidebar OFF
>   - Vertical tab OFF
>   - show sidebar (toolbar icon) OFF
> 
> >i know you will kill me about that 😅 
A. please add window bottoms (close,min,max) for mac
> B. Remove sideBERY header
> ![CleanShot 20240909 - 15-43](https://github.com/user-attachments/assets/d09fff00-5224-4e07-89f0-7a7ff1f5ab2c)



_________________________________________________________

> **photo [3]**
> 
>   - sideBERY ON
> Native nightly Firefox
>   - Sidebar ON
>   - Vertical tab OFF
>   - show sidebar (toolbar icon) ON
> ![CleanShot 20240909 - 15-56](https://github.com/user-attachments/assets/198c1ee6-8fc2-40ff-b40f-8b51d9db68ac)

**Reply by loaykhalifa** [reply]
I know this is alpha version, so i'm Sorry for too much photos and suggestion, that's because I am so excited for this project and I want you to take a look for [Zen browser](https://zen-browser.app/) they make sidebar side-by-side with vertical tab
![CleanShot 20240909 - 16-50](https://github.com/user-attachments/assets/bb9b5428-ef6a-4795-aa1b-4dc4b39ffa18)

**Reply by Naezr** [reply]
> A. please add window bottoms (close,min,max) for mac

Those buttons are now missing not only on the mac, but everywhere else in general. I've disabled them. Everything else needs to be finalized first, in order to implement those buttons there. They will come later. 

> B. Remove sideBERY header

Same thing with Sidebery. It's very easy to remove the sidebar header when Sidebery is open, it's a small thing. But first I need to finalize all logic to implement this properly.

**Reply by Naezr** [reply]
> they make sidebar side-by-side with vertical tab

Quite realizable, but only the width of the sidebar will be fixed and it will not be possible to change it by dragging over the edge.

-----------------------------------------------

Thanks for sharing about Zen Browser. They do things that I can't do with CSS alone. It's definitely a cool project and I'll be keeping an eye on it. However, in my opinion both Floorp and Zen Browser still feel pretty raw relative to vanilla Firefox. Zen looks neater, but there are still a lot of bugs and nasty rough edges.

But Zen has some cool animations, such as the popup animation. They can be ported to ShyFox. There's probably some other interesting stuff in there. I will dig into this project for interesting solutions that can be used in ShyFox.

**Comment by KAGEYAM4** [comment]
Does anybody know what is the behaviour of ArcBrowser when pin tabs increases, let's say 10. In sideberry the Ui is not usable if there are that many pinned items.  ( **Note** - i may be on v3.8.0 of ShyForx )
![image](https://github.com/user-attachments/assets/f0f4f708-25dc-464c-824a-548893ae6b35)

**Reply by Naezr** [reply]
Sidebery styles are not designed to use the navigation bar on the side. Tabs should be grouped three to a row

**Reply by KAGEYAM4** [reply]
> Sidebery styles are not designed to use the navigation bar on the side. Tabs should be grouped three to a row

ohh my bad. got it.


By the way have you tried vertical navigaton bar. I find it very useful as i have more than 10 panels and having navigation bar vertically seprates thing visually as we alaready have horizontal for tabs, pin - tabs .

**Comment by loaykhalifa** [comment]
I will use your theme as it is, so please please 
Tell me how to window bottoms (close, min, max) for mac
yes i know I will live with bug ,but i love your SHYFOX so much 
can't Wait until you done.
![CleanShot 20240911 - 23-59](https://github.com/user-attachments/assets/d4e8f747-123e-4778-b8c1-72e2a48c74e2)

**Reply by Naezr** [reply]
It's more complicated than you think. You can't just turn these buttons on

**Comment by loaykhalifa** [comment]
Disable hover after URL opened

https://github.com/user-attachments/assets/18db88aa-3d06-4f9a-af75-27c52f1d6b53

**Reply by Naezr** [reply]
I think it's already fixed

**Reply by Naezr** [reply]
`shyfox.appearance.leftbar.gapless`

try this after updating

**Reply by loaykhalifa** [reply]
👍 ... it fixed after i turn 
shyfox.position.extbutton ..... false 
shyfox.appearance.leftbar.gapless ... true

**Reply by claymorwan** [reply]
The layout looks fire tho how did u make it

**Comment by Naezr** [comment]
![image](https://github.com/user-attachments/assets/968c3ae1-27a9-44dd-ba75-a0e7d7cf5531)

**Comment by loaykhalifa** [comment]
Media Full-screen issue 
![CleanShot 2024-09-25 at 10 42 11](https://github.com/user-attachments/assets/ff4f4ae2-a918-4b9c-8638-2b55186f75c6)

**Reply by Naezr** [reply]
fixed

**Comment by loaykhalifa** [comment]
urlbar have too much icon 
![CleanShot 2024-09-25 at 11 03 17@2x](https://github.com/user-attachments/assets/b541bbd4-6748-43be-b8e9-6e580039d49a)

**Reply by semi710** [reply]
icons should be visible when the mouse is hovered.
and if the text is focused to edit then the icons should hide back again.

or a better approach would be to when url bar is focused increase it's width by like 80% and as soon as focus is removed back to it's place

**Reply by Naezr** [reply]
[Запись экрана от 2024-09-25 18-02-50.webm](https://github.com/user-attachments/assets/21f120a3-6c36-48d6-8cd8-80422d2c1d52)

**Reply by loaykhalifa** [reply]
All URL field disappear 
![CleanShot 2024-09-25 at 19 35 40](https://github.com/user-attachments/assets/c43b1343-fc99-4569-84ec-c5dd0d6c79b2)

**Reply by loaykhalifa** [reply]
Import this if you want to save some width in bookmark Toolbar
I know you can do something better

```

/* Kill bookmark folder icons in the Personal Toolbar */
toolbarbutton.bookmark-item > .toolbarbutton-icon {
  display: none !important;
}

```

**Reply by Naezr** [reply]
> All URL field disappear

Your Firefox version?

**Reply by loaykhalifa** [reply]
nightly; now working fine

**Comment by Naezr** [comment]
[Запись экрана от 2024-09-25 22-54-18.webm](https://github.com/user-attachments/assets/0cb173e0-3e2e-43d8-a812-411660d4972a)

**Comment by KAGEYAM4** [comment]
https://github.com/Naezr/ShyFox/issues/135#issuecomment-2354210466

Anyone having this lag in 4.0, i don't have 4.0. What about 3.8 stable ? 

just wanted to know if this lag can be improved in 4.0.

or is this lag only with me and something is wrong on my side?

**Comment by loaykhalifa** [comment]
I love what you did in last update 4.0 for url 
![CleanShot 2024-09-27 at 18 17 08](https://github.com/user-attachments/assets/12a089f4-26f7-497d-b88a-e7604454cdda)

There is a suggestion , you can put all icons of URL in (three dot icon) 
It will be much cleaner & instead of bookmarks icon (star icon) put reload icon 
to save space in upper navbar for windows control {close,min,max}

**Reply by Naezr** [reply]
https://addons.mozilla.org/firefox/addon/reload-in-address-bar/

Alas, it is impossible to create a drop-down menu for all these buttons using css styles alone.

**Comment by loaykhalifa** [comment]
Sidebar still open when popup window is open 
![CleanShot 2024-09-28 at 01 30 56@2x](https://github.com/user-attachments/assets/54fdd171-d2fb-4489-adcb-7d82c8dfd3a1)

**Reply by Naezr** [reply]
should be fixed now

**Comment by loaykhalifa** [comment]
Please, @Naezr I try many times for windows control, but I can't with 4.0 
could you please do that 
![CleanShot 2024-09-29 at 17 10 31@2x](https://github.com/user-attachments/assets/a638a8f0-f497-49be-8955-cb36c93badf8)

**Comment by Naezr** [comment]
[Запись экрана от 2024-10-02 12-21-27.webm](https://github.com/user-attachments/assets/19008686-0afb-4427-b3a6-ffc760bacb24)

**Comment by Naezr** [comment]
Nightly 133 just broke everything...

**Reply by Naezr** [reply]
ok, not everything

**Reply by Naezr** [reply]
I hope the developers will leave the option to leave the tab bar without the tabs themselves alone. Because now, when you enable vertical tabs, all buttons from the tab bar are raked into the navigation bar. And even if you move them manually back, they will all be moved to the navigation bar on restart.

**Comment by Naezr** [comment]
![image](https://github.com/user-attachments/assets/7ef9fea4-9665-4240-8fcb-f681b113c70e)

**Reply by QuantumKuba** [reply]
hot.

**Reply by loaykhalifa** [reply]
is that ready in 4.0 ?

**Comment by Halbour** [comment]
yooooooooo this left me so hype for 4.0!!
can't wait for the stable release...

**Comment by Naezr** [comment]
A little reflection:

I ran into the limitations of adequate CSS when creating 4.0, but was able to break through that wall with some complex tricks. However, along with this, CSS styles became very confusing and hard to read. At the same time, I started hitting the actual limitations of CSS, which can't be circumvented with tricks.

For this reason, I'm thinking of going back to fx-autoconfig to implement custom scripts, or even make my own Firefox fork. However, I can't pull off a fork just yet, since I'm alone and don't have enough knowledge to do it. So I will probably switch to fx-autoconfig in development 4.0. With this I will be able to do everything much easier, clearer, faster and at the same time many times cooler. If I can implement all of my ideas, it will be incredible.

In the meantime, I plan to continue supporting the current ShyFox so that browser updates don't break it.

**Comment by Neubulae** [comment]
Just a quick question: I didn't enjoy Firefox's own implementation of Sidebar so I decided to stick to the original Sideberry experience, plus I truly admire a "real borderless" experience(which removes visible border), so I need to tweak my own parameters. Would you include a blank file that is kept untouched(e.g.: `shy-user.css`) so people would whip up some of their own parameters in there?

**Reply by loaykhalifa** [reply]
+1
